File: Samu/Pr04/src/Cita.java

import java.util.*;
import java.text.SimpleDateFormat;

public class Cita {
    private static String BD_SERVER = "localhost";
    private static String BD_NAME = "GI1819";

    private int ID;
    private int NumSS;
    private Date Fecha_Hora;
    private String Consulta;

    public static List<Cita> ListaCitas(Paciente p) {
        List<Cita> lista = new ArrayList<>();
        BD miBD = new BD(BD_SERVER, BD_NAME);

        for (Object[] tupla : miBD.Select("SELECT * FROM tCita;")) {
            Cita u = new Cita((Integer) tupla[0]);
            lista.add(u);
        }
        return lista;
    }

    public Cita(int id) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        Object[] tupla = miBD.Select(String.format("Select * from tCita where ID_CITA=%d;", id)).get(0);
        this.ID = (Integer) tupla[0];
        this.NumSS = (Integer) tupla[1];
        this.Fecha_Hora = (Date) tupla[2];
        this.Consulta = (String) tupla[3];
    }

    public Cita(int ID_Paciente, Date Fecha, String consulta) {

        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
        String DateToStr = format.format(Fecha);
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Insert(String.format("insert into tCita(NumSSPaciente, Fecha_Hora, Consulta) " +
                "values (%d, '%s', '%s')", ID_Paciente, DateToStr, consulta));

        this.NumSS = ID_Paciente;
        this.Fecha_Hora = Fecha;
        this.Consulta = consulta;
    }

    public int getID() {
        return ID;
        // No se permite cambiar el ID de una Cita
    }

    public int getPacientePrueba() {
        return NumSS;
        // No se permite cambiar el paciente de una cita
    }

    public Date getFechaHora() {
        return Fecha_Hora;
    }

    public void setFechaHora(Date value) {
        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
        String DateToStr = format.format(value);
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Insert(String.format("update tCita set Fecha_hora = '%s' where ID_CITA=%d", DateToStr, this.ID));
        this.Fecha_Hora=value;
    }


    public String getConsulta() {
        return Consulta;
    }

    public void setConsulta(String value) {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Insert(String.format("update tCita set Consulta = '%s' where ID_CITA=%d", value, this.ID));
        this.Consulta=value;
    }

    public void BorrarCita() {
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Delete(String.format("Delete from tCita where ID_CITA = %d",this.ID));
    }

    public String toString() {
        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
        String DateToStr = format.format(Fecha_Hora);

        return this.ID + "/" + this.NumSS + "/" + DateToStr + "/" + this.Consulta;
    }
}

File: Aleks/HIS/Cita.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using BDLibrary;

namespace HIS
{
    public class Cita
    {
        private int id;
        private int numSS;
        private string consulta;
        private DateTime fecha_hora;


        private static string BD_SERVER = Properties.Settings.Default.BD_SERVER;
        private static string BD_NAME = Properties.Settings.Default.BD_NAME;

        public static List<Cita> ListaCitas(Paciente p)
        {
            List<Cita> lista = new List<Cita>();
            SQLSERVERDB miDB = new SQLSERVERDB(BD_SERVER, BD_NAME);
            foreach (object[] tupla in miDB.Select("SELECT id_cita FROM tCita where numsspaciente = " + p.NumeroSS_Paciente + ";"))
            {
                int id = (int)tupla[0];
                Cita c = new Cita(id);
                lista.Add(c);
            }
            return lista;
        }

        public Cita(int id)
        {
            SQLSERVERDB dbManager = new SQLSERVERDB(BD_SERVER, BD_NAME);
            object[] tupla = dbManager.Select("select * from tCita where id_cita = " + id + ";")[0];
            this.id = (int) tupla[0];
            this.numSS = (int) tupla[1];
            this.consulta = (string) tupla[3];
            this.fecha_hora = DateTime.Parse(tupla[2].ToString());
        }

        public Cita(int numSS, DateTime fecha_hora, string consulta)
        {
            SQLSERVERDB dbManager = new SQLSERVERDB(BD_SERVER, BD_NAME);
            dbManager.Insert("insert into tCita (numSSPaciente, fecha_hora, consulta) values"
                + "(" + numSS + ", '" + fecha_hora + "', '" + consulta + "');");
            this.id = (int) dbManager.Select("select max(id_cita) from tCita")[0][0];
            this.numSS = numSS;
            this.fecha_hora = fecha_hora;
            this.consulta = consulta;
        }

        public int ID
        {
            get { return id; }
        }

        public int NumSS
        {
            get { return numSS; }
            set
            {
                SQLSERVERDB dbManager = new SQLSERVERDB(BD_SERVER, BD_NAME);
                dbManager.Update("update tcita set numsspaciente = " + value + " where id_cita = "+ this.id + ";");
                numSS = value;
            }
        }

        public DateTime Fecha_Hora
        {
            get { return fecha_hora; }
            set
            {
                SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
                string sFecha = value.ToShortDateString() + " " + value.ToShortTimeString();
                miBD.Update("UPDATE tCita SET Fecha_Hora = '" + sFecha + "' WHERE ID_Cita=" + this.id + ";");
                fecha_hora = value;
            }
        }

        public string Consulta
        {
            get { return consulta; }
            set
            {
                SQLSERVERDB dbManager = new SQLSERVERDB(BD_SERVER, BD_NAME);
                dbManager.Update("update tcita set consulta = '" + value + "' where id_cita = " + this.id + ";");
                consulta = value;
            }
        }

        public void BorrarCita()
        {
            SQLSERVERDB dbManager = new SQLSERVERDB(BD_SERVER, BD_NAME);
            dbManager.Delete("delete tcita where id_cita = " + this.id + ";");
            this.numSS = -1;
            this.consulta = null;
            this.id = -1;
        }

        public override string ToString()
        {
            return ID + " " + NumSS + fecha_hora.ToShortDateString() + " " + fecha_hora.ToShortTimeString() + " " + consulta;
        }

        public override bool Equals(object obj)
        {
            return obj is Cita && ((Cita)obj).ID == ID;
        }

        public override int GetHashCode()
        {
            return ID.GetHashCode();
        }
    }
}
